api/src/main/java/io/opencensus/trace/CurrentSpanUtils.java

```diff
@@ -74,4 +100,73 @@ public void close() {
       }
     }
   }
+
+  private static final class RunnableInSpan implements Runnable {
+    private final Span span;
+    private final Runnable runnable;
+    private final boolean endSpan;
+
+    private RunnableInSpan(Span span, Runnable runnable, boolean endSpan) {
+      this.span = span;
+      this.runnable = runnable;
+      this.endSpan = endSpan;
+    }
+
+    @Override
+    public void run() {
+      Context origContext =
+          Context.current().withValue(ContextUtils.CONTEXT_SPAN_KEY, span).attach();
+      try {
+        runnable.run();
+      } catch (Throwable t) {
```

Comment:
if you catch (RuntimeException|Error e), then you don't need to use guava to throw

Reply:
You still want to catch any error to set the span status. If I don't use guava (because java6) the code will be like:

```
      try {
        runnable.run();
      } catch (Error e) {
        setErrorStatus(span, e);
        throw e;
      } catch (RuntimeException e) {
        setErrorStatus(span, e);
        throw e;
      } catch (Throwable t) {
        setErrorStatus(span, e);
        throw new RuntimeException("unexpected", t);
      } finally {
        Context.current().detach(origContext);
        if (endSpan) {
          span.end();
        }
      }
```

I think the current code is a bit nicer.